person.rb

```diff
@@ -0,0 +1,23 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name: 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
+
+  def can_use_services?
+    return true if @age >= 18 || @parent_permission == true
+  end
```

Comment:
- [ ] As stated the project requirement ` is_of_age?` must be Private method and `can_use_services?` must be Public method  but in **the case of yours `can_use_services?`  is Private method** because **Every instance method after private becomes a private method.**  So I recommend you to define all your public methods first, then define your private methods together at the end of the class.

Reply:
@Abenezer-Tilahun , thanks for point out and giving the detail suggestion to improve my code .

Reply:
Fixed.